### PR TITLE
Add a few TODOs, types, and an integration test

### PR DESCRIPTION
**Pull request by dragly**
These are points that should be revisited at some point.

### CODE REVIEW

File: f3df/src/renderables.rs [hunk 1]
```diff
@@ -87,6 +87,7 @@ pub fn convert_sector(sector: &crate::Sector) -> Sector {
             let i = cell_index_i_j % (cell_size[0]);
             let j = cell_index_i_j / (cell_size[0]);
             let k = cell_index / (cell_size_i_j);
+            // TODO this offset might be wrong
```
Comment (larsmoa) [inline f3df/src/renderables.rs:90]: Can you add some context to why you think so?

File: i3df/src/lib.rs [hunk 2]
```diff
@@ -136,7 +137,10 @@ pub fn parse_sector_header(mut input: &mut impl BufRead) -> Result<SectorHeader,
     // read attributes
     let attribute_count = input.read_u32::<LittleEndian>()?;
 
-    if attribute_count != 0 && attribute_count != ATTRIBUTE_COUNT {}
+    // TODO dragly 2020-04-22 should this not return an error?
+    if attribute_count != 0 && attribute_count != ATTRIBUTE_COUNT {
+
+    }
```
Comment (larsmoa) [inline i3df/src/lib.rs:143]: Shouldn't this jsn't be fixed right away?

Comment (dragly) [reply i3df/src/lib.rs]: Done

File: f3df/src/renderables.rs (hunk 1 above)
Comment (dragly) [reply f3df/src/renderables.rs]: Done

**Pull request merged**